### PR TITLE
Add Tiny JS

### PR DESCRIPTION
**Pull request by thoughtspile**
**https://github.com/thoughtspile/awesome-tiny-js#readme**

This is a list of small frontend libraries — mostly under 2Kb, with exceptions for select categories. Smaller frontend libraries help you build smaller apps, resulting in better user experience and fewer wasted resources. Apart from production usage: 
- A well-organized list of small libraries helps open-source developers focus their efforts on fields that are not already well-covered. 
- The cruft-free code of small libraries is an excellent learning material to understand how a class of libraries works — e.g. [hyperapp source](https://github.com/jorgebucaran/hyperapp/blob/main/index.js) is a great overview of virtual DOM.
- Many people put considerable effort into engineering size-conscious libraries, and deserve recognition for that.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

- [x] Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- [x] **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- [x] **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	- https://github.com/sindresorhus/awesome/pull/2569#issuecomment-1493308363
	- https://github.com/sindresorhus/awesome/pull/2565#issuecomment-1493303277
- [x] You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- [x] This pull request has a title in the format `Add Name of List`. It should not contain the word `Awesome`.
- [x] Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
- [x] Your entry should be added at the bottom of the appropriate category.
- [x] The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- [x] The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- [x] **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- [x] Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- [x] The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- [x] **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- [x] It's the result of hard work and the best I could possibly produce.
- [x] The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
- [x] The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
- [x] Non-generated Markdown file in a GitHub repo.
- [x] The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- [x] Not a duplicate. Please search for existing submissions.
- [x] Only has awesome items. Awesome lists are curations of the best, not everything.
- [x] Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- [x] Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- [x] Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- [x] Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
	- Should link back to this list.
- [x] Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- [x] Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- [x] Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- [x] All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- [x] Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- [x] Does not use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- [x] Does not include a CI (e.g. GitHub Actions) badge.<br>You can still use a CI for linting, but the badge has no value in the readme.
- [x] Does not include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

### CODE REVIEW

**Comment by thoughtspile** [comment]
unicorn

File: readme.md [hunk 1]
```diff
@@ -430,6 +430,7 @@
 - [MDBootstrap](https://github.com/mdbootstrap/awesome-mdbootstrap#readme) - Templates, layouts, components, and widgets to rapidly build websites.
 - [Master CSS](https://github.com/master-co/awesome-master-css#readme) - A virtual CSS language with enhanced syntax.
 - [Hydrogen](https://github.com/shopify/awesome-hydrogen#readme) - Edge-first framework for building Shopify storefronts with React.
+- [Tiny JS](https://github.com/thoughtspile/awesome-tiny-js#readme) - Frontend libraries smaller than 2Kb.
```
Comment (sindresorhus) [inline readme.md:433]: `Kb` is not a valid unit. I think you meant the SI unit `kB`.

Comment (sindresorhus) [reply readme.md:433]: Same applies to the descriptions in your list.

Comment (thoughtspile) [reply readme.md]: Thanks, shame on me. Should also be preceded by a space. Fixed everywhere.

**Comment by abdansyakuuro** [comment]
Wow

File: readme.md [hunk 2]
```diff
@@ -430,6 +430,7 @@
 - [MDBootstrap](https://github.com/mdbootstrap/awesome-mdbootstrap#readme) - Templates, layouts, components, and widgets to rapidly build websites.
 - [Master CSS](https://github.com/master-co/awesome-master-css#readme) - A virtual CSS language with enhanced syntax.
 - [Hydrogen](https://github.com/shopify/awesome-hydrogen#readme) - Edge-first framework for building Shopify storefronts with React.
+- [Tiny JS](https://github.com/thoughtspile/awesome-tiny-js#readme) - Frontend libraries smaller than 2 kB.
```
Comment (emmanuelgautier) [inline readme.md:433]: This is a great list. Maybe you can add the fact that the 2 kB includes all the dependencies making this list of libs more valuable.

Comment (kennethdost) [reply readme.md:433]: Thanl you

**Comment by Leonardbrett** [comment]
My first investment with Charlotte give me profits of over $7,000 and ever since then she has never failed to deliver and I can even say she's the must sincere broker I have ever known'' communicate with her to be a successful investor. On WhatsApp:+𝟭𝟰𝟳𝟱𝟮𝟰𝟮𝟲𝟯𝟯𝟴

**Comment by aabouzaid** [comment]
It's really nice and clean list.

I just have a minor suggestion about the file names in the repo.
The casing is inconsistent where:
- Some files are in uppercase, like: `README.md`.
- Some files are in lowercase, like `contributing.md `.

You can use any, but my suggestion is to keep them consistent.

**Review by andrzejsydor — APPROVED** [review]

**Comment by thoughtspile** [comment]
> It's really nice and clean list.

Thanks, that's good to hear!

> You can use any, but my suggestion is to keep them consistent.

Done, all lowercase now (casing changes are tricky in git)

**Comment by Swzzd** [comment]
image

**Comment by thoughtspile** [comment]
Hey @sindresorhus, is there anything else I could fix or improve here? Otherwise, maybe it's time to merge.

**Review by emmanuelgautier — APPROVED** [review]

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1659508021482127361

**Comment by thoughtspile** [comment]
Thanks, and have a nice weekend y'all ❤

**Pull request merged**